### PR TITLE
Explain enums and mention ternary-if in GDScript reference

### PR DESCRIPTION
**Pull request by bojidar-bg**
Comment if you feel like I'm missing something (or if the rst is wrong...)

### CODE REVIEW

File: reference/gdscript.rst [hunk 1]
```diff
@@ -516,6 +525,34 @@ expressions and must be assigned on initialization.
...
     const f = sin(20)  # sin() can be used in constant expressions
     const g = x + 20  # invalid; this is not a constant expression!
+    
+Enums
+^^^^^
+
+*Note, only available in Godot 2.2 or higher.*
+
+Enums are basically a shorthand for constants, and are pretty useful if you
+want to assing consecuitive integers to some constant.
```
Comment (akien-mga) [inline reference/gdscript.rst]: *assign, *consecutive

Comment (bojidar-bg) [inline reference/gdscript.rst]: ouch...

Comment (bojidar-bg) [inline reference/gdscript.rst]: Fixed!

File: reference/gdscript.rst [hunk 2]
```diff
@@ -516,6 +525,34 @@ expressions and must be assigned on initialization.
...
+    const TILE_FLOOR = 1
+    const TILE_SPIKE = 2
+    const TILE_TELEPORT = 3
+
+    enum State {STATE_IDLE, STATE_JUMP = 5, STATE_SHOOT}
+    # Is the same as:
+    const STATE_IDLE = 0
+    const STATE_JUMP = 5
+    const STATE_SHOOT = 6
+    const State = {STATE_IDLE = 0, STATE_JUMP = 5, STATE_SHOOT = 6}
```
Comment (Sslaxx) [inline reference/gdscript.rst]: This is a little confusing , I think. I assume the `const State` line is not meant to be bunched up with the `const STATE_` statements, so perhaps you're missing a `# Is also the same as:` type line?

Comment (vnen) [inline reference/gdscript.rst]: I think it is like this. You can use either `STATE_IDLE` or `State.STATE_IDLE`. Though this syntax wouldn't really work in GDScript, I believe it should be: 

```GDScript
const State = {"STATE_IDLE" = 0, "STATE_JUMP" = 5, "STATE_SHOOT" = 6}
```

Comment (bojidar-bg) [inline reference/gdscript.rst]: @vnen My syntax would work in GDScript... yours won't, except if you replace the `=` signs with `:` :D

Comment (vnen) [inline reference/gdscript.rst]: Oh, I stand corrected then. This quirk I didn't know.

Comment (akien-mga) [inline reference/gdscript.rst]: Yeah, GDScript supports both Python-style (`{"key": value}`) and Lua-style (`{key = value}`) dictionary declarations: http://docs.godotengine.org/en/latest/reference/gdscript.html#dictionary

**Pull request merged**